sql/charactersets.go

```diff
@@ -37,56 +37,55 @@ type CharacterSetsIterator struct {
 }
 
 // CharacterSetID represents a character set. Unlike collations, this ID is not intended for storage and may change as
-// new character sets are added. It is recommended to use the character set's name if persistence is desired.
+// the default collation changes. It is recommended to use the character set's name if persistence is desired.
 type CharacterSetID uint16
 
 // The character sets below are ordered alphabetically to make it easier to visually parse them.
 // As each ID acts as an index to the `characterSetArray`, they are explicitly defined.
-// It is recommended not to change any IDs when adding a new character set, as it will also require adjusting the array.
-// Instead, give it the next highest number, and insert it into the correct position alphabetically.
+// A character set's ID is defined as the default collation's ID.
 
 const (
-	CharacterSet_armscii8 CharacterSetID = 1
-	CharacterSet_ascii    CharacterSetID = 2
-	CharacterSet_big5     CharacterSetID = 3
-	CharacterSet_binary   CharacterSetID = 4
-	CharacterSet_cp1250   CharacterSetID = 5
-	CharacterSet_cp1251   CharacterSetID = 6
-	CharacterSet_cp1256   CharacterSetID = 7
-	CharacterSet_cp1257   CharacterSetID = 8
-	CharacterSet_cp850    CharacterSetID = 9
-	CharacterSet_cp852    CharacterSetID = 10
-	CharacterSet_cp866    CharacterSetID = 11
-	CharacterSet_cp932    CharacterSetID = 12
-	CharacterSet_dec8     CharacterSetID = 13
-	CharacterSet_eucjpms  CharacterSetID = 14
-	CharacterSet_euckr    CharacterSetID = 15
-	CharacterSet_gb18030  CharacterSetID = 16
-	CharacterSet_gb2312   CharacterSetID = 17
-	CharacterSet_gbk      CharacterSetID = 18
-	CharacterSet_geostd8  CharacterSetID = 19
-	CharacterSet_greek    CharacterSetID = 20
-	CharacterSet_hebrew   CharacterSetID = 21
-	CharacterSet_hp8      CharacterSetID = 22
-	CharacterSet_keybcs2  CharacterSetID = 23
-	CharacterSet_koi8r    CharacterSetID = 24
-	CharacterSet_koi8u    CharacterSetID = 25
-	CharacterSet_latin1   CharacterSetID = 26
-	CharacterSet_latin2   CharacterSetID = 27
-	CharacterSet_latin5   CharacterSetID = 28
-	CharacterSet_latin7   CharacterSetID = 29
-	CharacterSet_macce    CharacterSetID = 30
-	CharacterSet_macroman CharacterSetID = 31
-	CharacterSet_sjis     CharacterSetID = 32
-	CharacterSet_swe7     CharacterSetID = 33
-	CharacterSet_tis620   CharacterSetID = 34
+	CharacterSet_armscii8 CharacterSetID = 32
```

Comment:
Probably prefer to sort these by their ids rather than alphabetically

Reply:
For the most part, the IDs are sort of arbitrary (except for identification purposes), and they can change. They seem to only be used in a super-small section of the MySQL protocol, and it seems like most clients ignore it considering we've not run into any issues while returning incorrect values so far. It's far more intuitive to look them up though when they're sorted alphabetically (or some other human-relatable way).